### PR TITLE
feat: Add Text leaflet layer

### PR DESCRIPTION
**Pull request by jbrundage**
Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

**Comment by jbrundage** [comment]
image

File: packages/map/src/leaflet/Text.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+Text.prototype.publish("text", "", "string", "Default text");
+Text.prototype.publish("textColumn", null, "set", "Font awesome column", function () { return this.columns(); }, { optional: true });
+Text.prototype.publish("textColor", "#ffffff", "html-color", "Default  font awesome Color");
+Text.prototype.publish("textColorColumn", null, "set", "Text color column", function () { return this.columns(); }, { optional: true });
+Text.prototype.publish("textFontFamily", null, "set", "Text color column", function () { return this.columns(); }, { optional: true });
+Text.prototype.publish("strokeColor", "transparent", "html-color", "Default stroke Color", null, { optional: true });
+Text.prototype.publish("strokeColorColumn", null, "set", "Stroke color column", function () { return this.columns(); }, { optional: true });
+Text.prototype.publish("fillColor", "#376cea", "html-color", "Default fill Color", null, { optional: true });
+Text.prototype.publish("fillColorColumn", null, "set", "Fill color column", function () { return this.columns(); }, { optional: true });
+Text.prototype.publish("fontFamily", "Verdana", "string", "Font family", null, { optional: true });
```
Comment (GordonSmith) [inline packages/map/src/leaflet/Text.ts:98]: Should default to null and be optional (thus default to page settings)

File: packages/map/src/leaflet/Text.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            let text = this.propValue(textIdx, row, this.text());
+            const textSize = this.textSize(text, this.fontFamily(), this.fontSize());
+            if (this.textFormat_exists()) {
+                text = d3Format(this.textFormat())(text);
+            }
+            return {
+                icon: BeautifyIcon({
+                    iconShape: "rectangle",
+                    // isAlphaNumericIcon: true,
+                    // text: this.propValue(textIdx, row, this.text()),
```
Comment (GordonSmith) [inline packages/map/src/leaflet/Text.ts:27]: Remove commented code

File: apps/arjuna_map/index.html [hunk 3]
```diff
@@ -0,0 +1,523 @@
...
+                    "West Virginia"
+                ],
+                [
+                    "WY",
+                    "43.475968",
+                    "-108.09028",
+                    "Wyoming"
+                ]
+            ]
+        }
```
Comment (GordonSmith) [inline apps/arjuna_map/index.html:519]: Could this become an option of the US States Choropleth?

File: packages/map/src/leaflet/Text.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                    textColor: this.propValue(textColorIdx, row, this.textColor()),
+                    borderColor: this.propValue(strokeColorIdx, row, this.strokeColor()),
+                    backgroundColor: this.propValue(fillColorIdx, row, this.fillColor()),
+                    props: {
+                        owner: this,
+                        row
+                    }
+                }),
+                draggable: false
+            };
```
Comment (GordonSmith) [inline packages/map/src/leaflet/Text.ts:47]: Something similar could be achieved without the BeutifyIcon (and would be "lighter")?

File: apps/arjuna_map/index.html (hunk 3 above)
Comment (jbrundage) [reply apps/arjuna_map/index.html:519]: Sure, I'll create an issue for this.

File: packages/map/src/leaflet/Text.ts (hunk 4 above)
Comment (jbrundage) [reply packages/map/src/leaflet/Text.ts:47]: Wouldn't this require changing core classes for a very minor performance benefit? As it is this is very similar to `Pins.ts`.

**Pull request merged**